net/host.go

```diff
@@ -183,16 +183,25 @@ func (h *Host) CloseProtocolStream(offerID types.Hash) {
 	h.swapMu.RLock()
 	swap, has := h.swaps[offerID]
 	h.swapMu.RUnlock()
-	if !has {
+	if !has || swap.streamClosed {
 		return
 	}
 
+	swap.streamClosed = true
 	log.Debugf("closing stream: peer=%s protocol=%s",
 		swap.stream.Conn().RemotePeer(), swap.stream.Protocol(),
 	)
+
 	_ = swap.stream.Close()
 }
 
+// DeleteOngoingSwap deletes an ongoing swap from the network's state.
+func (h *Host) DeleteOngoingSwap(offerID types.Hash) {
+	h.swapMu.Lock()
+	defer h.swapMu.Unlock()
+	delete(h.swaps, offerID)
```

Comment:
Do we need to be closing the stream if it is not already closed here?

Reply:
the caller of this function always closes the stream as well, I can make a note of that in the function comment

Reply:
Can we ether add a panic (if we are certain it can never happen) or an error level log if the assertion in the comment turns out to be false at some point? That way if we introduce a slow resource leak in the future, it will get detected quickly.